DigitalLearningSolutions.Data/DataServices/ProgressDataService.cs

```diff
@@ -227,5 +229,31 @@ public void UnlockProgress(int progressId)
                 new { progressId }
             );
         }
+
+        public IEnumerable<LearningLogEntry> GetLearningLogEntries(int progressId)
+        {
+            return connection.Query<LearningLogEntry>(
+                @"SELECT 
+                            ses.LoginTime AS [When],
+                            ses.Duration  AS LearningTime,
+                            NULL AS AssessmentTaken,
+                            NULL AS AssessmentScore,
+                            NULL AS AssessmentStatus
+                        FROM [Sessions] AS ses 
+                        INNER JOIN Progress AS pr ON pr.CustomisationID = ses.CustomisationID AND pr.CandidateID = ses.CandidateID
+                        WHERE pr.ProgressID = @progressId
+                    UNION ALL SELECT 
+                            aa.[Date] AS [When],
+                            NULL AS Duration,
+                            sec.SectionName AS AssessmentTaken,
+                            aa.Score AS AssessmentScore,
+                            aa.[Status] AS AssessmentStatus
+                        FROM AssessAttempts AS aa 
+                        INNER JOIN dbo.Customisations AS cu ON cu.CustomisationID = aa.CustomisationID
+                        LEFT JOIN Sections AS sec ON sec.ApplicationID = cu.ApplicationID AND sec.SectionNumber = aa.SectionNumber
+                        WHERE aa.ProgressID = @progressId",

```

Comment:
Formatting appears a little strange in this SQL query are there a mix of tabs and spaces?

Reply:
Yeah I think it's better with the UNION ALL on a separate line, almost would be nice to have a way to do it like
```
    SELECT ...
UNION
    SELECT ...
```
but I think that looks even stranger.